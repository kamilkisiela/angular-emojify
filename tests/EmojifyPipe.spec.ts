import { Pipe } from '@angular/core';

import './__common';
import { EmojifyPipe } from '../src';

describe('EmojifyPipe', () => {
  const pipe: Pipe = Reflect.getOwnMetadata('annotations', EmojifyPipe)[0];
  let emojifyPipe: EmojifyPipe;
  
  beforeEach(() => {
    emojifyPipe = new EmojifyPipe();
  });
  
  it('should has transform method', () => {
    expect(emojifyPipe.transform).toBeDefined();
  });

  it('should transform', () => {
    const text = `It's a :rocket:!`;
    const mocked = `mocked: ${text}`;
    
    expect(emojifyPipe.transform(text)).toBe(mocked);
  });

  it('should be named emojify', () => {
    expect(pipe.name).toBe('emojify');
  });

  it('should be pure', () => {
    expect(pipe.pure).toBe(true);
  });
});
