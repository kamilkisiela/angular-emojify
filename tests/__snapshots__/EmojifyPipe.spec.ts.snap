// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmojifyPipe should transform 1`] = `"It's a ðŸš€!"`;
